src/interfaces/permissions/request/message.ts

```diff
@@ -0,0 +1,135 @@
+import type { AuthCreateOpts, Authorizable, AuthVerificationResult } from '../../../messages/types';
+import type { JsonPermissionsRequest, PermissionsRequestDescriptor } from './types';
+import type { Scope, Conditions } from '../types';
+
+import { DIDResolver } from '../../../did/did-resolver';
+import { GeneralJwsSigner, GeneralJwsVerifier } from '../../../jose/jws/general';
+import { generateCid, parseCid } from '../../../utils/cid';
+import lodash from 'lodash';
+import { Message } from '../../../messages/message';
+import { v4 as uuidv4 } from 'uuid';
+
+const { isPlainObject } = lodash;
+
+type PermissionsRequestOpts = AuthCreateOpts & {
+  conditions?: Conditions;
+  description: string;
+  grantedTo: string;
+  grantedBy: string;
+  objectId?: string;
+  scope: Scope;
+};
+
+export class PermissionsRequest extends Message implements Authorizable {
+  protected message: JsonPermissionsRequest;
+
+  constructor(message: JsonPermissionsRequest) {
+    super(message);
+  }
+
+  get conditions(): Conditions {
+    return this.message.descriptor.conditions;
+  }
+
+  get grantedBy(): string {
+    return this.message.descriptor.grantedBy;
+  }
+
+  get grantedTo(): string {
+    return this.message.descriptor.grantedTo;
+  }
+
+  get description(): string {
+    return this.message.descriptor.description;
+  }
+
+  get scope(): Scope {
+    return this.message.descriptor.scope;
+  }
+
+  static getType(): string {
+    return 'PermissionsRequest';
+  }
+
+  static getInterface(): string {
+    return 'Permissions';
+  }
+
+  static async create(opts: PermissionsRequestOpts): Promise<PermissionsRequest> {
+    const { conditions } = opts;
+    const providedConditions = conditions ? conditions : {};
+    const mergedConditions = { ...DEFAULT_CONDITIONS, ...providedConditions  };
+
+    const descriptor: PermissionsRequestDescriptor = {
+      conditions  : mergedConditions,
+      description : opts.description,
+      grantedTo   : opts.grantedTo,
+      grantedBy   : opts.grantedBy,
+      method      : 'PermissionsRequest',
+      objectId    : opts.objectId ? opts.objectId : uuidv4(),
+      scope       : opts.scope,
+    };
+
+    const descriptorCid = await generateCid(descriptor);
+
+    const authPayload = { descriptorCid: descriptorCid.toString() };
+    const authPayloadStr = JSON.stringify(authPayload);
+    const authPayloadBytes = new TextEncoder().encode(authPayloadStr);
+
+    const signer = await GeneralJwsSigner
+      .create(authPayloadBytes, [opts.signingMaterial]);
+
+    const message: JsonPermissionsRequest = { descriptor, authorization: signer.getJws() };
+
+    return new PermissionsRequest(message);
+  }
+
+  /**
+   * @throws {Error} if descriptorCid is missing from Auth payload
+   */
+  async verifyAuth(didResolver: DIDResolver): Promise<AuthVerificationResult> {
```

Comment:
Design discussion: do you think it might be worth while consider singleton for DID resolver so that many methods/interfaces can be less bloated.

Reply:
definitely down for that. I think it should be tackled in a separate PR that addresses `DIDResolver`

Reply:
Cool, created an issue to track: https://github.com/TBD54566975/dwn-sdk-js/issues/61